### PR TITLE
Add tab plugin.

### PR DESCRIPTION
**Pull request by bobthecow**
This is a port of a bash function I wrote almost five years ago:

  https://gist.github.com/bobthecow/757788

The fish version is better though ;)

I only have OS X with iTerm and Terminal to test on, but it should be possible to add support for other terminals on other OS. Send a pull request!

### CODE REVIEW

File: plugins/tab/tab.fish [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#     tab [PATH]            Open PATH in a new tab
+#     tab [CMD]             Open a new tab and execute CMD
+#     tab [PATH] [CMD] ...  You can prolly guess
+
+function tab -d 'Open the current directory (or any other directory) in a new tab'
+  set -l cmd ''
+  set -l cdto (pwd)
+
+  if test (count $argv) -gt 0
+    pushd . >/dev/null
```
Comment (bucaran) [inline plugins/tab/tab.fish]: This pushes the current directory to the directory stack right? But why?

Comment (bobthecow) [inline plugins/tab/tab.fish]: Yep :)

Comment (bobthecow) [inline plugins/tab/tab.fish]: That way I let fish resolve relative paths and symlinks and all that for me, rather than trying to sort it out myself.

Comment (bobthecow) [inline plugins/tab/tab.fish]: Updated this whole block in 6c85e8d to make it a bit clearer what's happening.

File: plugins/tab/tab.fish [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#     tab [CMD]             Open a new tab and execute CMD
+#     tab [PATH] [CMD] ...  You can prolly guess
+
+function tab -d 'Open the current directory (or any other directory) in a new tab'
+  set -l cmd ''
+  set -l cdto (pwd)
+
+  if test (count $argv) -gt 0 -a -d $argv[1]
+    pushd . >/dev/null
+    cd $argv[1]
```
Comment (bucaran) [inline plugins/tab/tab.fish]: What if `$argv[1]` is an invalid directory?

Comment (bobthecow) [inline plugins/tab/tab.fish]: that's what `-a -d $argv[1]` is supposed to check.

Comment (bucaran) [inline plugins/tab/tab.fish]: Right.

**Comment by bucaran** [comment]
Looks good. 

@bpinto May cavil at the lack of tests, but other than that, updating the plugins README and we are good to go.

**Comment by bobthecow** [comment]
I added it to the plugins README and squashed commits :)

**Comment by bucaran** [comment]
:+1:

File: plugins/tab/tab.fish [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#
+# Usage:
+#     tab                   Opens the current directory in a new tab
+#     tab [PATH]            Open PATH in a new tab
+#     tab [CMD]             Open a new tab and execute CMD
+#     tab [PATH] [CMD] ...  You can prolly guess
+
+function tab -d 'Open the current directory (or any other directory) in a new tab'
+  set -l cmd ''
+  set -l cdto (pwd)
```
Comment (bpinto) [inline plugins/tab/tab.fish]: Probably better to use `$PWD` instead, so we don't start another process?

Comment (bobthecow) [inline plugins/tab/tab.fish]: Works for me. Fixed in aec321b

**Comment by bpinto** [comment]
Awesome, thanks for this plugin @bobthecow !

**Comment by bobthecow** [comment]
No problem!

**Pull request merged**